### PR TITLE
optimization store in fullCSC

### PR DESCRIPTION
**Pull request by GalaxySciTech**
- optimization store in fullCSC
- store receiptHash in fullCSC
- optimization error log

### CODE REVIEW

File: contracts/FullCheckpoint.sol [hunk 1]
```diff
@@ -64,26 +64,21 @@ contract FullCheckpoint {
...
-        (bytes32 ph1, int256 n1, uint64 rn) = HeaderReader.getBlock1Params(
-            block1Header
-        );
-        require(n == 0 && n1 == 1, "Invalid Init Block");
+
+        HeaderReader.ValidationParams memory block1 = HeaderReader
+            .getValidationParams(block1Header);
+        require(n == 0 && block1.number == 1, "Invalid Init Block");
         headerTree[genesisHeaderHash] = Header({
+            receiptHash: bytes32(0),
```
Comment (wgr523) [inline contracts/FullCheckpoint.sol]: is it really `block0.receiptHash = bytes32(0)`?

File: contracts/FullCheckpoint.sol [hunk 2]
```diff
@@ -279,16 +275,20 @@ contract FullCheckpoint {
     }
 
     function setCommittedStatus(bytes32 startBlock) internal {
```
Comment (wgr523) [inline contracts/FullCheckpoint.sol:279]: Is committed and finalized same? If so, please add a comment for this function

Comment (GalaxySciTech) [reply contracts/FullCheckpoint.sol:279]: finalized==false equals mainnetNum==-1

File: contracts/FullCheckpoint.sol (hunk 1 above)
Comment (GalaxySciTech) [reply contracts/FullCheckpoint.sol]: fix

File: contracts/FullCheckpoint.sol [hunk 3]
```diff
@@ -278,17 +270,25 @@ contract FullCheckpoint {
         }
     }
 
+    /* @dev Confirm all ancestor unconfirmed block
```
Comment (wgr523) [inline contracts/FullCheckpoint.sol]: can you explain to readers that the word "confirm"="commit"="finalize" ?

Comment (GalaxySciTech) [reply contracts/FullCheckpoint.sol]: sure

**Review by wgr523 — APPROVED** [review]

**Pull request merged**